pandas/core/frame.py

```diff
@@ -9239,15 +9250,50 @@ def stack(self, level: IndexLabel = -1, dropna: bool = True, sort: bool = True):
         dog kg     2.0     NaN
             m      NaN     3.0
         """
-        from pandas.core.reshape.reshape import (
-            stack,
-            stack_multiple,
-        )
+        if not v3:
```

Comment:
Should there be a FutureWarning in this path?

Reply:
Ref: the top of https://github.com/pandas-dev/pandas/pull/53921#issuecomment-1613818826